src/ore/src/assert.rs

```diff
@@ -289,6 +289,148 @@ macro_rules! assert_contains {
     }};
 }
 
+/// Asserts that the provided expression, that returns an `Option`, is `None`.
+///
+/// # Motivation
+///
+/// The standard pattern for asserting a value is `None` using the `assert!` macro is:
+///
+/// ```
+/// # let x: Option<usize> = None;
+/// assert!(x.is_none());
+/// ```
+///
+/// The issue with this pattern is when the assertion fails it only prints `false`
+/// and not the value contained in the `Some(_)` variant which makes debugging difficult.
+///
+/// # Examples
+///
+/// ### Basic Use
+///
+/// ```should_panic
+/// use mz_ore::assert_none;
+/// assert_none!(Some(42));
+/// ```
+///
+/// ### With extra message
+///
+/// ```should_panic
+/// use mz_ore::assert_none;
+/// let other_val = 100;
+/// assert_none!(Some(42), "ohh noo! x {other_val}");
+/// ```
+///
+#[macro_export]
+macro_rules! assert_none {
+    ($val:expr, $($msg:tt)+) => {{
+        if let Some(y) = &$val {
+            let formatted_msg = format!($($msg)+);
+            panic!("assertion failed: expected None found Some({y:?}), {formatted_msg}");
```

Comment:
Move the `formatted_msg` into the `panic` so it'll be outlined, here and below.

Reply:
Done!